.travis.yml

```diff
@@ -29,6 +29,11 @@ before_deploy:
   - pip install --user awscli
 
 deploy:
+  - provider: script
+    skip_cleanup: true
+    script:
+      npx semantic-release
```

Comment:
Hmm... I think this file needs some re-jigging:

The `# deploy master to Staging` block should have its `on: branch: master` condition changed to be the new `develop` branch. I.e. every commit to develop will deploy to the staging env. Obviously the comment at the top of the block should be updated to reflect this.

Similarly, the `# deploy tags matching rel-* or release-* to Production` should be changed to simply trigger on commits to `master` rather than commits that were tagged with a SemVer. Once that's been done, I'd suggest simply adding the `npx semantic-release` in there - immediately above the S3 deployment commands.

If I'm not mistaken that will then create the behaviour we now desire:

* Commits to `develop`: Build, test & deploy to stanging env
* Commits to `master`: Build, test, **run semantic release** & deploy to production env

Reply:
@james-nash Updated as requested. Do we still need the cron job titled `# deploy master branch to Production once a day`?

Reply:
Thanks for updating those branch conditions.

Will the `provider: script` block that contains `npx semantic-release` always run since it has no `on: ...` condition at the end? I'm not sure - I tried looking at the Travis docs briefly but couldn't find a clear answer.
If that is what'll happen though, should we not move that command into the one for deploying master merges since those are the only ones that count as "releases"? So, something like this:

```yaml
  # deploy master merges to production
  - provider: script
    script:
      npx semantic-release &&
      ~/.local/bin/aws s3 sync dist s3://${PROD_BUCKET} --delete --region=${PROD_BUCKET_REGION} --exclude "*" --include "*.html" --content-type "text/html; charset=utf-8" &&
      ~/.local/bin/aws s3 sync dist s3://${PROD_BUCKET} --delete --region=${PROD_BUCKET_REGION} --exclude "*" --include "*.css" --content-type "text/css; charset=utf-8" &&
      ~/.local/bin/aws s3 sync dist s3://${PROD_BUCKET} --delete --region=${PROD_BUCKET_REGION} --exclude "*" --include "*.js" --content-type "application/javascript; charset=utf-8" &&
      ~/.local/bin/aws s3 sync dist s3://${PROD_BUCKET} --delete --region=${PROD_BUCKET_REGION} --include "*" --exclude "*.js" --exclude "*.html" --exclude "*.css" &&
      ~/.local/bin/aws cloudfront create-invalidation --distribution-id ${PROD_DISTRIBUTION_ID} --paths /*
    skip_cleanup: true
    on:
branch: master
```

Perhaps @dm knows the answer?

In any case, to answer your question: Yes, we do need to cron job bit. That's there to ensure our careers page is kept reasonably up-to-date. The job listings that appear there are fetched as part of the site build. So, by scheduling a rebuild & redeploy of the website once a day, we ensure that the job listings are never more than 24h out of date.

Reply:
So the release command only runs on `master` by default. Best practice is to always run it. The log just reports it as "Only runs on master, skip" essentially. This is good because it verifies it still runs to some degree with the CI and gives it a separate log chunk. You can whitelist other branches to run semantic release if you want (not recommended though).

So to answer your question. It's in the best possible place to run as things stand. No need to combine it with the other command. Will help create a cleaner log keeping it out of there I believe. Currently running semantix-release this way in three other repos and appears to be working great.

Reply:
Ah ok. I didn't realise that `semantic-release` checked the branch itself as well.

In that case, let's do this! :-D